### DISCUSSION TITLE
Seaborn in vaex

### DISCUSSION THREAD

**Discussion by dharik-arsath** [vaexio/vaex#1407] [Q&A]
How to use Seaborn in vaex dataframes.

**Comment by JovanVeljanoski** [comment]
As far as I can see, Seaborn is implemented assuming pandas dataframes as input. 

So for the foreseeable future, I don't see how one can use Seaborn with vaex. If somebody wants to make a new package like Seaborn centred around vaex that would be a different matter. 

Since the APIs of pandas and vaex are not exactly matching, it would be quite a chore to make a package like Seaborn to support both dataframes.. 

So for now, your best bet is to do all the aggregations etc.. with vaex, and then use matplotlib or plotly and the like to display the results. It is not as convenient as using seaborn - i know.

**Reply by dharik-arsath** [reply]
To overcome all the confusion between pandas and vaex, It will be nice if you "Take up a project using pandas code and do exact replica in Vaex" so the conversion from pandas to vaex will be lot easier for people. Thanks.

**Reply by dharik-arsath** [reply]
Hi, Plotly is not working with vaex dataframe I did px.histogram(data_frame=data,x=colname) and I tried this as well px.histogram(x=data.colname) in both cases I get different error. In first case this error is displayed ===>  "Value of 'x' is not the name of a column in 'data_frame'. Expected one of [0] but received: MonthlyCharges"

In the next case where I did px.histogram(x=data.colname) I get : list indices must be integers or slices, not str. If I convert into pandas df by using .to_pandas_df and then plot with plotly it is working. Please look into this issue

**Reply by JovanVeljanoski** [reply]
My exact same answer from seaborn applies to plotly express. 
You should read both the documantation of vaex and plotly express.
